### PR TITLE
[master] Porting #49195 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #49195 to master

### CODE REVIEW

**Comment by mchugh19** [comment]
This is a dependency on porting #49395 and #53068

File: salt/modules/saltcheck.py [hunk 1]
```diff
@@ -112,55 +139,41 @@ def run_test(**kwargs):
         return "Test must be a dictionary"
 
 
-def run_state_tests(state):
+def run_state_tests(state, saltenv=None, check_all=False):
     '''
     Execute all tests for a salt state and return results
     Nested states will also be tested
 
     :param str state: the name of a user defined state
```
Comment (dhiltonp) [inline salt/modules/saltcheck.py]: We need a docstring for saltenv, too

File: tests/unit/modules/test_saltcheck.py [hunk 2]
```diff
@@ -321,6 +322,42 @@ def test__assert_less_equal3(self):
             mybool = sc_instance._SaltCheck__assert_less_equal(aaa, bbb)
             self.assertEqual(mybool, 'Pass')
 
+    def test__assert_empty(self):
```
Comment (dhiltonp) [inline tests/unit/modules/test_saltcheck.py]: It's great that we have additional tests here.

Maybe I'm missing something, but I'm not seeing tests for the new runtests functionality (saltenv, checkall).

**Review by dhiltonp — CHANGES_REQUESTED** [review]

File: salt/modules/saltcheck.py (hunk 1 above)
Comment (dhiltonp) [reply salt/modules/saltcheck.py]: Also, I think the docstring for `state` is now out of date.

File: tests/unit/modules/test_saltcheck.py (hunk 2 above)
Comment (mchugh19) [reply tests/unit/modules/test_saltcheck.py]: This is part of a series of updates for the saltcheck module, the latest of which is #53068. Might it make sense to allow these ports into master, and handle the doc and test updates as part of the most recent PR?

Comment (dhiltonp) [reply tests/unit/modules/test_saltcheck.py]: Maybe?

Should #53068 to point at master?

**Comment by mchugh19** [comment]
I've ported these changes and the other neon saltcheck updates into #55613

**Comment by waynew** [comment]
There's a

**Pull request closed without merging**